### PR TITLE
Add support for remmaping arguments

### LINKED ISSUE #288: Allow to configure topic names in tango_ros_node
When integrating tango_ros_node in another rosjava application (i.e. Tangobot) it is currently not possible to configure topic names, remappings or namespaces.

it would be very valuable to extend tango_ros_node to take in topic remapping arguments in general from the rosjava side and pass them through to the native node

### PR DESCRIPTION
**Pull request by PerrineAguiar**
Fixes #288.

### CODE REVIEW

**Comment by PerrineAguiar** [comment]
@jubeira @adamantivm FYI
I tested it but it seems that there is an error on the rosjava side, I opened a PR in the rosjava_core repo to fix this: https://github.com/rosjava/rosjava_core/pull/248.

**Comment by adamantivm** [comment]
Thank you @PerrineAguiar ! Please let me know when releasing a new public version of rosjava core becomes blocking for this or a new TRS release.

File: TangoRosStreamer/tango_nodelet_manager/src/main/jni/jni.cc [hunk 1]
```diff
@@ -57,9 +57,31 @@ Java_eu_intermodalics_nodelet_1manager_TangoNodeletManager_execute(
...
+          jstring remap_to_value = (jstring) (env->GetObjectArrayElement(remapping_objects_value, i * 2 + 1));
+          const char* remap_to = env->GetStringUTFChars(remap_to_value, NULL);
+
+          remappings.insert(std::pair<std::string, std::string>(std::string(remap_from), std::string(remap_to)));
+
+          env->ReleaseStringUTFChars(remap_from_value, remap_from);
+          env->ReleaseStringUTFChars(remap_to_value, remap_to);
+        }
+      } else {
+        LOG(WARNING) << "Wrong remapping arguments array, its size should be an even number but is "
```
Comment (mcopejans) [inline TangoRosStreamer/tango_nodelet_manager/src/main/jni/jni.cc]: Make this a LOG(ERROR) and return in this case. Motivation: if the array is not empty, the user wants to use the remapping feature. So it is not used correctly, this should be made clear.

File: TangoRosStreamer/tango_nodelet_manager/src/main/jni/jni.cc [hunk 2]
```diff
@@ -57,9 +57,31 @@ Java_eu_intermodalics_nodelet_1manager_TangoNodeletManager_execute(
...
+          << remappingStringCount << ". Remapping will not be apply.";
+      }
+    } else {
+      LOG(WARNING) << "Remapping arguments array is null. Remapping will not be apply.";
```
Comment (mcopejans) [inline TangoRosStreamer/tango_nodelet_manager/src/main/jni/jni.cc]: LOG(INFO) << No topic remapping to be done.
Do not put implementation details in the logging.

**Comment by PerrineAguiar** [comment]
@mcopejans Thanks for your review. I addressed your comments and change the code to follow the [ROS remapping syntax](http://wiki.ros.org/Remapping%20Arguments).

**Comment by mcopejans** [comment]
LGTM

**Comment by PerrineAguiar** [comment]
This PR was waiting for [this fix](https://github.com/rosjava/rosjava_core/pull/248) on the rosjava side.
A [new version of rosjava_core](https://github.com/rosjava/rosjava_mvn_repo/pull/44) which includes the fix has been released.
After testing that the remapping feature works as expected, I am merging this PR.

**Pull request merged**